cpp/log/cluster_state_controller-inl.h

```diff
@@ -296,7 +314,15 @@ void ClusterStateController<Logged>::OnClusterStateUpdated(
         // own class, and share an instance between the interested
         // parties.
         all_peers_.emplace(node_id, peer);
-        fetcher_->AddPeer(node_id, peer);
+
+        // Only matching versions are added to fetcher so we don't talk
+        // to nodes that have a different version
+        if (!node_version_mismatch) {
+          fetcher_->AddPeer(node_id, peer);
+        } else {
+          VLOG(1) << "Node has mismatched version, dropping: " << node_id;
```

Comment:
I wish I'd started this on Rietveld now. Probably too late.

Martin

On 1 December 2015 at 10:33, Al Cutter <notifications@github.com> wrote:

> In cpp/log/cluster_state_controller-inl.h
> <https://github.com/google/certificate-transparency/pull/1071#discussion_r46262613>
> :
>
> > @@ -296,7 +314,15 @@ void ClusterStateController<Logged>::OnClusterStateUpdated(
> >          // own class, and share an instance between the interested
> >          // parties.
> >          all_peers_.emplace(node_id, peer);
> > -        fetcher_->AddPeer(node_id, peer);
> > +
> > +        // Only matching versions are added to fetcher so we don't talk
> > +        // to nodes that have a different version
> > +        if (!node_version_mismatch) {
> > +          fetcher_->AddPeer(node_id, peer);
> > +        } else {
> > +          VLOG(1) << "Node has mismatched version, dropping: " << node_id;
>
> Should probably promote that to a WARNING?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/certificate-transparency/pull/1071/files#r46262613>
> .
>